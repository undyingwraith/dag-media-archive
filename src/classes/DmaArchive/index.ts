export * from './DmaArchive'
export * from './IDmaArchive'
