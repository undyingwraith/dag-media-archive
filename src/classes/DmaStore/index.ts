export * from './IDmaStore'
export * from './DmaStore'
