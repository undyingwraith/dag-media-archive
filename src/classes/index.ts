export * from './DmaArchive'
export * from './DmaStore'
