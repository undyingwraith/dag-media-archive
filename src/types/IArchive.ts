import CID from 'cids';

export interface IArchive {
	name: string
	meta?: CID
	media: CID
	version: number
}
