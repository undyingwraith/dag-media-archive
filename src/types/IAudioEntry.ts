import CID from 'cids';
import {IMediaEntry} from './IMediaEntry';

export interface IAudioEntry extends IMediaEntry {
	cover?: CID
}
