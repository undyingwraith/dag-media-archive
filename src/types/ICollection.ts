import CID from 'cids';

export interface ICollection {
	name: string
	entries: CID
}
