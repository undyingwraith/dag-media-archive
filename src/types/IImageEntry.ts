import CID from 'cids';
import {IMediaEntry} from './IMediaEntry';

export interface IImageEntry extends IMediaEntry {
	thumbnail: CID
}
