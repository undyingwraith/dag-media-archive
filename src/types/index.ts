export * from './IMediaEntry'
export * from './IArchive'
