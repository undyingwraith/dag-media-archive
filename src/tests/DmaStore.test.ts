import {DmaStore, IDmaStore} from '../classes';
import CID from 'cids';
import Ipfs, {IPFS} from 'ipfs';
import {IMediaEntry} from '../types';
import {CONSTANTS} from './const';


describe('DmaStore write', () => {
	let node: IPFS;
	let store: IDmaStore<IMediaEntry>;

	beforeAll(async () => {
		node = await Ipfs.create();
	}, 10000);

	beforeEach(async () => {
		store = await DmaStore.create(node, CONSTANTS.baseCid);
	}, 10000);

	it('init works', async () => {
		await expect(DmaStore.initialize(node)).resolves.toStrictEqual(CONSTANTS.baseCid);
	});

	it('write works', async () => {
		await expect(store.writeMedia({
			id: 'ebb941ae-419e-4848-8664-57597ddb214e',
			source: new CID('bafybeia2fz5fkkqmjlivj7jrlecschnkkhy2z5ced3velgpjydl2lzdglu'),
		})).resolves.toStrictEqual(CONSTANTS.baseCidWOne);
	});

	it('consecutive write works', async () => {
		await expect(store.writeMedia({
			id: 'ebb941ae-419e-4848-8664-57597ddb214e',
			source: new CID('bafybeia2fz5fkkqmjlivj7jrlecschnkkhy2z5ced3velgpjydl2lzdglu'),
		})).resolves.toStrictEqual(CONSTANTS.baseCidWOne);
		await expect(store.writeMedia({
			id: '5c14df51-3d66-4389-849f-92aa04ffff61',
			source: new CID('bafybeidbnognu6nvv2z7dmmtqqpqtb47ya6d3loiqsrofadxtgg7yblbui'),
		})).resolves.toStrictEqual(CONSTANTS.baseCidWTwo);
	});
});

describe('DmaStore read', () => {
	let node: IPFS;
	let store: IDmaStore<IMediaEntry>;

	beforeAll(async () => {
		node = await Ipfs.create();
	}, 10000);

	beforeEach(async () => {
		store = await DmaStore.create(node, CONSTANTS.baseCidWTwo);
	}, 10000);

	it('read works', async () => {
		await expect(store.readMedia('5c14df51-3d66-4389-849f-92aa04ffff61')).resolves.toMatchSnapshot();
	});

	it('getList works', async () => {
		await expect(store.getList()).resolves.toMatchSnapshot();
	});
});
