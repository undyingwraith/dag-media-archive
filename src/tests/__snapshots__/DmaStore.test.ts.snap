// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DmaStore init works 1`] = `
Object {
  "media": Object {
    "codec": "dag-cbor",
    "hash": Object {
      "data": Array [
        20,
        64,
        112,
        72,
        79,
        96,
        233,
        165,
        80,
        68,
        120,
        155,
        221,
        18,
        138,
        251,
        38,
        224,
        177,
        44,
        145,
        11,
        13,
        35,
        157,
        20,
        97,
        128,
        228,
        198,
        228,
        87,
        99,
        116,
        64,
        201,
        42,
        24,
        3,
        63,
        238,
        85,
        238,
        184,
        20,
        170,
        121,
        99,
        37,
        0,
        207,
        59,
        218,
        85,
        157,
        151,
        208,
        78,
        103,
        136,
        74,
        179,
        136,
        8,
        19,
        148,
      ],
      "type": "Buffer",
    },
    "version": 1,
  },
  "name": "DmaStore",
  "version": 1,
}
`;

exports[`DmaStore read works 1`] = `
Object {
  "id": "5c14df51-3d66-4389-849f-92aa04ffff61",
  "source": Object {
    "codec": "dag-pb",
    "hash": Object {
      "data": Array [
        18,
        32,
        97,
        107,
        140,
        218,
        121,
        181,
        174,
        179,
        241,
        177,
        147,
        132,
        31,
        9,
        135,
        159,
        192,
        60,
        61,
        173,
        200,
        132,
        162,
        226,
        128,
        119,
        153,
        141,
        252,
        5,
        97,
        162,
      ],
      "type": "Buffer",
    },
    "version": 1,
  },
}
`;
